pygments/lexers/bdd.py

```diff
@@ -0,0 +1,131 @@
+"""
+    pygments.lexers.bdd
+    ~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for BDD(Behavior-driven development). 
+    More information: https://en.wikipedia.org/wiki/Behavior-driven_development
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, include, bygroups
+from pygments.token import Comment, Keyword, Name, String, Number, Text, Punctuation, Whitespace
+
+__all__ = ['BddLexer']
+
+class BddLexer(RegexLexer):
+    """
+    Lexer for BDD(Behavior-driven development), which highlights not only keywords, 
+    but also comments, punctuations, strings, numbers, and variables.
+
+    .. versionadded:: 2.10
+    """
+
+    name = 'Bdd'
+    aliases = ['bdd']
+    filenames = ['*.feature']
+    mimetypes = ['text/x-bdd']
+    
+    # pre-defined keywords
+    feature_keywords = '^(:)()$'
+    feature_element_keywords = '^(\\s*)()(:)$'
+    examples_keywords = '^(\\s*)()(:)$'
+    step_keywords = '^(\\s*)(Given|When|Then|Add|And|Feature|Scenario Outline|Scenario|Background|Examples|But)'
+
+    tokens = {
+        'comments': [
+            (r'^\s*#.*$', Comment),
+        ],
+        'feature_elements': [
+            (step_keywords, Keyword, "step_content_stack"),
+            include('comments'),
+            (r"(\s)", Whitespace),
```

Comment:
Any reason for using a group match here? Also maybe consider capturing multiple spaces for efficiency?
```suggestion
            (r"\s+", Whitespace),
```

Reply:
Thank you very much for your patient guidance. I overcomplicated the problem. I delete the unnecessary parts. Now it is more concise. 😃